### ISSUE TITLE
When bootstrapping a new node - not all data is streamed 

### ISSUE THREAD

**Issue by slivne** [scylladb/scylla#2797]
*Installation details*
Scylla version (or git commit hash): 2.0.rc4
- I was trying to search the failure and ended up being on the following commit:
8ca1432b04428022de3eda3d3d85fdf4d4bcc3ef 
with this applied:

```
diff --git a/main.cc b/main.cc
index 7e82d9a..37d1d73 100644
--- a/main.cc
+++ b/main.cc
@@ -623,7 +623,7 @@ int main(int ac, char** av) {
             engine().at_exit([&cf_cache_hitrate_calculator] { return cf_cache_hitrate_calculator.stop(); });
             cf_cache_hitrate_calculator.local().run_on(engine().cpu_id());
             supervisor::notify("starting native transport");
-            gms::get_local_gossiper().wait_for_gossip_to_settle();
+            gms::get_local_gossiper().wait_for_gossip_to_settle().get();
             api::set_server_gossip_settle(ctx).get();
             supervisor::notify("starting cf cache hit rate calculator");
             service::get_local_storage_service().start_native_transport().get();
```
and also the patch from avi "[PATCH scylla v1] storage_proxy: scan more nodes exponentially to achieve target result set size"

the thing is that the test is not passing for other reasons (and we altered the test to make sure it will pass)

I need to test the previous commit to see if it will pass or not.

currently testing without avi's patch to see if it can be "blammed".
Cluster size: 50

with ```--smp 2 --memory 1G```
update_cluster_layout_tests.py:TestLargeScaleCluster.add_50_nodes_test

We are seeing that the test ends with some data missing after the 49th/50th node is added.

it seems the issue is that the 50th node does not stream from all the nodes it needs to stream data from 

according to the ring the token for the missing data 

(I created an additional ks.table with the same settings)

```
cqlsh> select token(key),key from ks1.cf;

 system.token(key)    | key
----------------------+---------
 -4017745849639943611 | k198066

(1 rows)
cqlsh> select * from ks.cf where key='k198066';
```

this value was inserted by node49 (after it was bootstrapped - succesfuly) - and we know it was on node46 and was possible to see it before (the code checks that)

checking node46 sstables with sstabledump (after doing a flush) - also shows the value existed when it owned it
```
./node46/data/ks/cf-ce212770987e11e7acf2000000000001/ks-cf-ka-2-Data.db
[
  {
    "partition" : {
      "key" : [ "k198066" ],
      "position" : 0
    },
    "rows" : [
      {
        "type" : "row",
        "position" : 93,
        "liveness_info" : { "tstamp" : "2017-09-13T13:49:25.415735Z" },
        "cells" : [
          { "name" : "c1", "value" : "value1" },
          { "name" : "c2", "value" : "value2" }
        ]
      }
    ]
  },
```

according to the ring the ownership should have moved to node50

```
127.0.0.35  rack1       Up     Normal  3.88 MB         ?                  -4017758778153559415                        
127.0.0.50  rack1       Up     Normal  3.2 MB          ?                   -4017281356590317108                        
127.0.0.46  rack1       Up     Normal  3.43 MB         ?                   -4015615366541287479  
```
complete ring info 
[ring.txt](https://github.com/scylladb/scylla/files/1300193/ring.txt)

yet looking at node50 logs it did not stream from node46

[node50.log.txt](https://github.com/scylladb/scylla/files/1300194/node50.log.txt)

trying to check that node50 sees all the cluster (not after bootstrap completed - now) shows that its aware of all the nodes

```
sh-4.3$ /data2/shlomi/scylla-tools-java/bin/nodetool -p 12000 status
Datacenter: datacenter1
=======================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address     Load       Tokens       Owns    Host ID                               Rack
UN  127.0.0.1   8.64 MB    256          ?       faa5392f-72b5-4fbc-8ca7-3056961f3cd6  rack1
UN  127.0.0.2   8.22 MB    256          ?       87fed88d-a707-4bed-993d-6797fb457f10  rack1
UN  127.0.0.3   8.18 MB    256          ?       db7ffc92-7199-40fe-b974-15bb0953d9d5  rack1
UN  127.0.0.4   9.05 MB    256          ?       a0716945-117f-4562-85c8-4fb354d54bbb  rack1
UN  127.0.0.5   8.29 MB    256          ?       d5439c95-c16b-4b5a-b509-ee5e8cda84ef  rack1
UN  127.0.0.6   8.62 MB    256          ?       cd55ec6a-11ce-40b6-9909-4cbd54e21212  rack1
UN  127.0.0.7   7.61 MB    256          ?       bc3e35fd-0a1b-46af-9d17-a4417bed732e  rack1
UN  127.0.0.8   7.91 MB    256          ?       cfde653b-7302-44a3-abdc-f686d4af6f73  rack1
UN  127.0.0.9   7.33 MB    256          ?       81bc8680-53fd-4ac9-821a-b120ab18ca46  rack1
UN  127.0.0.10  6.97 MB    256          ?       bea61c23-6067-4786-b5f9-cdff0f4acce1  rack1
UN  127.0.0.11  7.32 MB    256          ?       784c0d7a-d174-4caa-a2f2-31aa73b46b3a  rack1
UN  127.0.0.12  6.51 MB    256          ?       9d4493fc-8b23-4229-baa4-5dd9d522137d  rack1
UN  127.0.0.13  6.57 MB    256          ?       8a10860f-5064-433d-85a1-5e9c9a2850d2  rack1
UN  127.0.0.14  6.88 MB    256          ?       090c2342-5daf-477f-b14b-da4ebfe0e31d  rack1
UN  127.0.0.15  6.82 MB    256          ?       b0afd282-2b14-47b2-a553-716ff7ee01de  rack1
UN  127.0.0.16  6.14 MB    256          ?       63d9ae70-fca8-4880-b57d-37e466082408  rack1
UN  127.0.0.17  6.17 MB    256          ?       0986fda7-3a28-4ec9-850a-4f26214fc022  rack1
UN  127.0.0.18  6.09 MB    256          ?       2123b177-a9a0-4c45-b885-337971be045c  rack1
UN  127.0.0.19  6.15 MB    256          ?       d50fee37-b8ac-4999-9abb-83a72e022e5d  rack1
UN  127.0.0.20  5.7 MB     256          ?       3ec2ba44-a1ec-43a5-ad7a-3445976a6800  rack1
UN  127.0.0.21  5.93 MB    256          ?       d415fd1c-ec9b-4536-9e60-45c2ec733942  rack1
UN  127.0.0.22  5.33 MB    256          ?       fd84d51d-dfc4-44ba-9cc4-2a49b1371d11  rack1
UN  127.0.0.23  5.43 MB    256          ?       951685fb-91fb-4fd6-9839-b8db9c1759ee  rack1
UN  127.0.0.24  5.03 MB    256          ?       d686665a-7be2-45bf-8c5f-3f644b4c0ee6  rack1
UN  127.0.0.25  5.11 MB    256          ?       1ef02bbf-e385-475b-b961-cad15dafe602  rack1
UN  127.0.0.26  4.84 MB    256          ?       3710eed2-9200-48d4-a660-24d73b1da748  rack1
UN  127.0.0.27  4.96 MB    256          ?       d06161e6-970b-4777-901a-9bb7f8193348  rack1
UN  127.0.0.28  4.46 MB    256          ?       855aa9e1-12f8-40e4-ae0b-40772f958b4a  rack1
UN  127.0.0.29  4.45 MB    256          ?       71ae8ad3-8ab9-4302-ba8e-8a9d2643b8b4  rack1
UN  127.0.0.30  3.76 MB    256          ?       65a4aff6-0d61-43f4-b512-d304e505dcdb  rack1
UN  127.0.0.31  4.61 MB    256          ?       c50c4ab9-ee21-4047-b62c-0ff13841a17c  rack1
UN  127.0.0.32  4.04 MB    256          ?       0433f8ef-8694-4897-b9f7-6a781abc5907  rack1
UN  127.0.0.33  4.29 MB    256          ?       97314a82-dd92-44c9-bf65-ba5829207cd9  rack1
UN  127.0.0.34  3.77 MB    256          ?       b8a1edc4-ff62-4938-bc48-87a4e89b7eb5  rack1
UN  127.0.0.35  3.91 MB    256          ?       ed9d751a-a597-4499-8b06-23043ebb8ade  rack1
UN  127.0.0.36  3.69 MB    256          ?       fad7d5c1-4ec1-4c2e-b923-23b2962d6ed6  rack1
UN  127.0.0.37  3.92 MB    256          ?       71c7117d-2338-4fbf-90dd-b35f5e6b6d6a  rack1
UN  127.0.0.38  3.63 MB    256          ?       e25b9fad-a76e-4a89-9467-a25a8cfef069  rack1
UN  127.0.0.39  3.82 MB    256          ?       96541268-6609-4a0f-a437-b49d7c7eb2f7  rack1
UN  127.0.0.40  3.63 MB    256          ?       9a7ab9f9-9c6c-4ef2-9d6c-83d2aec3f968  rack1
UN  127.0.0.41  3.74 MB    256          ?       f08aee09-ebb8-456e-b7a7-905e4dcda5ce  rack1
UN  127.0.0.42  3.65 MB    256          ?       d0b7591a-0731-49d1-b2a5-339373bb598e  rack1
UN  127.0.0.43  3.61 MB    256          ?       fec9077f-4929-4271-a42a-28ca168f4928  rack1
UN  127.0.0.44  3.34 MB    256          ?       8ca0f22e-4e1b-4df5-ad1f-7d9ce6544322  rack1
UN  127.0.0.45  3.29 MB    256          ?       c397b067-eea8-4543-8c96-1e45befd5371  rack1
UN  127.0.0.46  3.46 MB    256          ?       b39fe7fc-8e1b-456c-8f45-aad9b1923aab  rack1
UN  127.0.0.47  3.19 MB    256          ?       09a475f4-a10f-4d88-ad0d-57ba9c53b8dc  rack1
UN  127.0.0.48  3.26 MB    256          ?       3336049e-91f8-48e8-a582-b2fd2143291f  rack1
UN  127.0.0.49  3.19 MB    256          ?       3c852e13-7ce6-431a-946c-a1725efd14ef  rack1
UN  127.0.0.50  3.24 MB    256          ?       4732b4f9-e10b-40a6-9d50-7e71f42808bf  rack1

Note: Non-system keyspaces don't have the same replication settings, effective ownership information is meaningless
```

it also has the same token ownership info using nodetool ring and specifically the questioned range it should have streamed

```
127.0.0.35  rack1       Up     Normal  3.91 MB         ?                  -4017758778153559415                        
127.0.0.50  rack1       Up     Normal  3.24 MB         ?                  -4017281356590317108                        
127.0.0.46  rack1       Up     Normal  3.46 MB         ?                   -4015615366541287479       
```

**Comment by slivne** [comment]
after removing Avi's patch the test passes 

so @asias - you are off the hook avi we need to re-look at your patch.

I am closing this issue - what a goose chase - and will send a patch to fix the test till we have the  1230 fixed

**Comment by gleb-cloudius** [comment]
After Avi's patch scanning should be much faster, so it may complete
before things are settled down (it should not start before they did, but
may be this is the bug here). I wouldn't close it so fast unless your
findings are incorrect.

--
			Gleb.

**Comment by slivne** [comment]
it does not seem related to avi's patch ... after trying multiple times ...

using 8567762339774ead73e9e786ef988cde009a0d44

the test failes (this should include some streaming changes)

in the test we do not stream to all destinations and it seems that we halt "building the streamers" on first occurrence of a remote node not being accessible

cluster 1
```
.
.
.
INFO  2017-09-16 03:12:54,690 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Starting streaming to 127.0.0.37
INFO  2017-09-16 03:12:54,690 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Beginning stream session with 127.0.0.37
INFO  2017-09-16 03:12:54,690 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Starting streaming to 127.0.0.38
INFO  2017-09-16 03:12:54,690 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Beginning stream session with 127.0.0.38
INFO  2017-09-16 03:12:54,690 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Starting streaming to 127.0.0.39
INFO  2017-09-16 03:12:54,690 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Beginning stream session with 127.0.0.39
INFO  2017-09-16 03:12:54,691 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Starting streaming to 127.0.0.40
INFO  2017-09-16 03:12:54,691 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Beginning stream session with 127.0.0.40
INFO  2017-09-16 03:12:54,691 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Starting streaming to 127.0.0.41
INFO  2017-09-16 03:12:54,691 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Beginning stream session with 127.0.0.41
INFO  2017-09-16 03:12:54,691 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Starting streaming to 127.0.0.42
INFO  2017-09-16 03:12:54,691 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Beginning stream session with 127.0.0.42
INFO  2017-09-16 03:12:54,691 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Starting streaming to 127.0.0.43
INFO  2017-09-16 03:12:54,691 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Beginning stream session with 127.0.0.43
INFO  2017-09-16 03:12:54,698 [shard 0] gossip - InetAddress 127.0.0.43 is now DOWN, status = NORMAL
INFO  2017-09-16 03:12:54,707 [shard 0] rpc - client 127.0.0.43: client connection dropped: connection is closed
INFO  2017-09-16 03:12:54,710 [shard 0] messaging_service - Retry verb=15 to 127.0.0.43:0, retry=10: seastar::rpc::closed_error (connection is closed)
INFO  2017-09-16 03:12:54,710 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Prepare completed with 127.0.0.4. Receiving 7, sending 0
INFO  2017-09-16 03:12:54,710 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Prepare completed with 127.0.0.1. Receiving 7, sending 0
INFO  2017-09-16 03:12:54,710 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Prepare completed with 127.0.0.7. Receiving 7, sending 0
INFO  2017-09-16 03:12:54,710 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Prepare completed with 127.0.0.9. Receiving 7, sending 0
INFO  2017-09-16 03:12:54,710 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Prepare completed with 127.0.0.8. Receiving 7, sending 0
INFO  2017-09-16 03:12:54,710 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Prepare completed with 127.0.0.12. Receiving 7, sending 0
INFO  2017-09-16 03:12:54,711 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Prepare completed with 127.0.0.11. Receiving 7, sending 0
INFO  2017-09-16 03:12:54,711 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Prepare completed with 127.0.0.10. Receiving 7, sending 0
INFO  2017-09-16 03:12:54,711 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Prepare completed with 127.0.0.19. Receiving 7, sending 0
```

cluster2
```
.
.
.
INFO  2017-09-17 00:43:55,872 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Starting streaming to 127.0.0.33
INFO  2017-09-17 00:43:55,872 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Beginning stream session with 127.0.0.33
INFO  2017-09-17 00:43:55,872 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Starting streaming to 127.0.0.34
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Beginning stream session with 127.0.0.34
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Starting streaming to 127.0.0.35
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Beginning stream session with 127.0.0.35
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Starting streaming to 127.0.0.36
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Beginning stream session with 127.0.0.36
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Starting streaming to 127.0.0.37
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Beginning stream session with 127.0.0.37
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Starting streaming to 127.0.0.38
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Beginning stream session with 127.0.0.38
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Starting streaming to 127.0.0.39
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Beginning stream session with 127.0.0.39
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Starting streaming to 127.0.0.40
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Beginning stream session with 127.0.0.40
INFO  2017-09-17 00:43:55,873 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Starting streaming to 127.0.0.41
INFO  2017-09-17 00:43:55,874 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Beginning stream session with 127.0.0.41
INFO  2017-09-17 00:43:55,875 [shard 0] gossip - InetAddress 127.0.0.40 is now DOWN, status = NORMAL
INFO  2017-09-17 00:43:55,898 [shard 0] rpc - client 127.0.0.40: client connection dropped: connection is closed
INFO  2017-09-17 00:43:55,900 [shard 0] messaging_service - Retry verb=15 to 127.0.0.40:0, retry=10: seastar::rpc::closed_error (connection is closed)
INFO  2017-09-17 00:43:55,900 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Prepare completed with 127.0.0.6. Receiving 7, sending 0
INFO  2017-09-17 00:43:55,900 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Prepare completed with 127.0.0.3. Receiving 7, sending 0
INFO  2017-09-17 00:43:55,901 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Prepare completed with 127.0.0.4. Receiving 7, sending 0
INFO  2017-09-17 00:43:55,901 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Prepare completed with 127.0.0.5. Receiving 7, sending 0
INFO  2017-09-17 00:43:55,901 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Prepare completed with 127.0.0.7. Receiving 7, sending 0
INFO  2017-09-17 00:43:55,901 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Prepare completed with 127.0.0.11. Receiving 7, sending 0
INFO  2017-09-17 00:43:55,901 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Prepare completed with 127.0.0.9. Receiving 7, sending 0
INFO  2017-09-17 00:43:55,901 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Prepare completed with 127.0.0.14. Receiving 7, sending 0
INFO  2017-09-17 00:43:55,901 [shard 0] stream_session - [Stream #224abd10-9b28-11e7-b4f6-000000000000] Prepare completed with 127.0.0.10. Receiving 7, sending 0
```
[cluster2_node48.log.txt](https://github.com/scylladb/scylla/files/1309065/cluster2_node48.log.txt)
[cluster1_node49.log.txt](https://github.com/scylladb/scylla/files/1309066/cluster1_node49.log.txt)

please note I enabled the boot_strapper logging so the ranges of the data that are missing are included in the request to build streamers

**Comment by asias** [comment]
In cluster 1, we saw a failed streaming verb, but after retrying, the verb completed successfully. In the end, the streaming for bootstrap also finished successfully.

```
INFO  2017-09-16 03:12:54,691 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Starting streaming to 127.0.0.43
INFO  2017-09-16 03:12:54,691 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Beginning stream session with 127.0.0.43
INFO  2017-09-16 03:12:54,698 [shard 0] gossip - InetAddress 127.0.0.43 is now DOWN, status = NORMAL
INFO  2017-09-16 03:12:54,707 [shard 0] rpc - client 127.0.0.43: client connection dropped: connection is closed
INFO  2017-09-16 03:12:54,710 [shard 0] messaging_service - Retry verb=15 to 127.0.0.43:0, retry=10: seastar::rpc::closed_error (connection is closed)

...

INFO  2017-09-16 03:13:24,724 [shard 0] messaging_service - Retry verb=15 to 127.0.0.43:0, retry=9: OK
INFO  2017-09-16 03:13:24,724 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Prepare completed with 127.0.0.43. Receiving 7, sending 0

INFO  2017-09-16 03:13:27,381 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] Session with 127.0.0.43 is complete, state=COMPLETE
INFO  2017-09-16 03:13:27,384 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] bytes_sent = 0, bytes_received = 3364236, tx_bandwidth = 0.000 MiB/s, rx_bandwidth = 0.091 MiB/s
INFO  2017-09-16 03:13:27,384 [shard 0] stream_session - [Stream #c7a42770-9a73-11e7-8246-000000000000] All sessions completed for streaming plan Bootstrap, peers={127.0.0.1, 127.0.0.2, 127.0.0.3, 127.0.0.4, 127.0.0.5, 127.0.0.6, 127.0.0.7, 127.0.0.8, 127.0.0.9, 127.0.0.10, 127.0.0.11, 127.0.0.12, 127.0.0.13, 127.0.0.14, 127.0.0.15, 127.0.0.16, 127.0.0.17, 127.0.0.18, 127.0.0.19, 127.0.0.20, 127.0.0.21, 127.0.0.22, 127.0.0.23, 127.0.0.24, 127.0.0.25, 127.0.0.26, 127.0.0.27, 127.0.0.28, 127.0.0.29, 127.0.0.30, 127.0.0.31, 127.0.0.32, 127.0.0.33, 127.0.0.34, 127.0.0.35, 127.0.0.36, 127.0.0.37, 127.0.0.38, 127.0.0.39, 127.0.0.40, 127.0.0.41, 127.0.0.42, 127.0.0.43}

```

**Comment by slivne** [comment]
@asias 
cluster 1 bootstrapped the 49th node.

In the bootstrap process of that node (49):
*  we started to stream information from nodes 1-43 and then had the error (that was later retired and succeeded)
* we did not stream information from nodes 44-48

To be clear we needed to ...

**Comment by asias** [comment]
Looking at the logs, at the time we start streaming, the joining node hasn't managed to apply all the endpoint_state for peer nodes, explained in #2404, the the joining node does not know some of the nodes yet. 

In #2404, we know gossip applies the endpoint_state in the ip address order, this aligns with what we saw in the log, we streamed from the first 1-44 nodes .

All sessions completed for streaming plan Bootstrap, peers={127.0.0.1, 127.0.0.2, 127.0.0.3, 127.0.0.4, 127.0.0.5, 127.0.0.6, 127.0.0.7, 127.0.0.8, 127.0.0.9, 127.0.0.10, 127.0.0.11, 127.0.0.12, 127.0.0.13, 127.0.0.14, 127.0.0.15, 127.0.0.16, 127.0.0.17, 127.0.0.18, 127.0.0.19, 127.0.0.20, 127.0.0.21, 127.0.0.22, 127.0.0.23, 127.0.0.24, 127.0.0.25, 127.0.0.26, 127.0.0.27, 127.0.0.28, 127.0.0.29, 127.0.0.30, 127.0.0.31, 127.0.0.32, 127.0.0.33, 127.0.0.34, 127.0.0.35, 127.0.0.36, 127.0.0.37, 127.0.0.38, 127.0.0.39, 127.0.0.40, 127.0.0.41, 127.0.0.42, 127.0.0.43, 127.0.0.44}

In addition to port the fixes to #2404,  d2632ddf1dfb718b3246a325aa44ffe8c27641fe (Merge "gossip: optimize apply_state_locally for large cluster" from Asias) we need to send the echo message async.
